### ISSUE TITLE
Missing algorithms - SGD modified_huber

### ISSUE THREAD

**Issue by writetokavitha** [dotnet/machinelearning#3843]
@codemzs 
I am from the marketing automation team at MSFT and doing intent detection on lead emails. We use the SGD and SVM in sci-kit learn in production and find it missing in ML.Net. Could you look into adding
SGD with following loss function:
loss="hinge": (soft-margin) linear Support Vector Machine,
loss="modified_huber": smoothed hinge loss,
loss="log": logistic regression,

We are getting 10%+ points increase in accuracy using modified_huber rather than hinge or log loss function. All the loss functions would be important for us priority would be
1. Modified_huber
2. hinge
3. log

https://scikit-learn.org/stable/modules/sgd.html#classification

**Comment by yaeldekel** [comment]
Hi @writetokavitha, ML.NET has an implementation of SDCA (stochastic dual coordinate ascent) for linear classification. This algorithm supports the three types of loss you specified above. For example, you can specify which type of loss you want like this:

```
mlContext.BinaryClassification.Trainers.SdcaNonCalibrated(labelColumnName: "Label", featureColumnName: "Features", lossFunction: new HingeLoss());
```
See the two APIs to create an SDCA trainer:
[here](https://github.com/dotnet/machinelearning/blob/master/src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs#L255) and [here](https://github.com/dotnet/machinelearning/blob/master/src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs#L281).

**Comment by ganik** [comment]
Closing this as @yaeldekel replied with info on loss funcitons
